raft/node.go

```diff
@@ -314,13 +319,27 @@ func (n *node) run(r *raft) {
 		// TODO: maybe buffer the config propose if there exists one (the way
 		// described in raft dissertation)
 		// Currently it is dropped in Step silently.
-		case m := <-propc:
+		case pm := <-propc:
+			m := pm.m
 			m.From = r.id
-			r.Step(m)
-		case m := <-n.recvc:
+			err := r.Step(m)
+			if pm.result != nil {
+				if err == ErrProposalDropped {
```

Comment:
why do we only propagate `ErrProposalDropped` error? we should propagate back all types of errors.